test/test_residency_implementation_handoff.py

```
@@ -716,5 +769,613 @@ def test_phase_append_commit_changes_only_the_manifest(self) -> None:
+            list(TASK_020_FIXTURE_PATHS),
+            "task_evidence[TASK-020]",
+        )
+
+    def _assert_invalid(
+        self,
+        metadata_mutator: Callable[[dict[str, object]], None],
+        expected: str,
+    ) -> None:
+        _, metadata, _ = self._inputs()
+        metadata_mutator(metadata)
+        with self.assertRaises(HandoffError) as raised:
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)